### PR TITLE
Update free-programming-books-langs.md

### PR DESCRIPTION
**Pull request by Khushi-Shukla**
## What does this PR do?
Adds resource

### Description
This PR adds a book on React which is JavaScript library. The book is suitable for beginner who are interested in learning React.

### Why is this valuable (or not)?
The book is divided into lessons and uses simple language to explain important concepts. This book can also be considered as a 30 days routine for people who want to start their journey in React.

### How do we know it's really free?
The link provided directly opens the pdf which can be downloaded on any device of choice.

### For book lists, is it a book? For course lists, is it a course? etc.
This is a book.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1288,6 +1288,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 #### React
 
+* [30 days of React: An introduction to React in 30 bite-size morsels](https://www.newline.co/fullstack-react/assets/media/sGEMe/MNzue/30-days-of-react-ebook-fullstackio.pdf) (PDF)
```
Comment (shrn01) [inline books/free-programming-books-langs.md:1291]:
```suggestion
* [30 days of React: An introduction to React in 30 bite-size morsels](https://www.newline.co/fullstack-react/assets/media/sGEMe/MNzue/30-days-of-react-ebook-fullstackio.pdf) - Ari Lerner (PDF)
```
Please provide author attribution

**Review by shrn01 — CHANGES_REQUESTED** [review]

**Comment by Khushi-Shukla** [comment]
Please merge this PR or add the label of ready to merge @shrn01

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by Khushi-Shukla** [comment]
Thank you so much. Please add the "hacktoberfest-accepted" label.

**Pull request merged**